### PR TITLE
feat: Add new variable for allowing ECR image sharing to lambda service in external account

### PR DESCRIPTION
**Pull request by thepoppingone**
This is extensible to other services like codebuild, however I only require lambda image to be shared to the lambda service of another for now

## Description
Introduce another variable to allow ARN of lambda functions ARNs to be stated

## Motivation and Context
Currently, the module hard codes the principal value to "AWS" only. Hence I am not able to use it for other services despite this option of inserting services to be included the policy statement being present in the AWS console
More docs here:
https://aws.amazon.com/blogs/compute/introducing-cross-account-amazon-ecr-access-for-aws-lambda/

Also this module consolidates all statements into a single policy, I cannot work around it but to add to it to handle for lambda case instead.

## Breaking Changes
None

## How Has This Been Tested?
It uses a variable with a default value of [] that will not trigger the dynamic statement

### CODE REVIEW

**Review by antonbabenko — CHANGES_REQUESTED** [review]
Looks good, but please revert unnecessary changes to files in the `.github` folder.

File: main.tf [hunk 1]
```diff
@@ -62,6 +62,33 @@ data "aws_iam_policy_document" "repository" {
     }
   }
 
+
+  dynamic "statement" {
+    for_each = var.repository_type == "private" && var.repository_lambda_read_access_arns != [] ? [1] : []
```
Comment (bryantbiggs) [inline main.tf:67]:
```suggestion
    for_each = var.repository_type == "private" && length(var.repository_lambda_read_access_arns) > 0 ? [1] : []
```

**Comment by thepoppingone** [comment]
image

**Review by antonbabenko — APPROVED** [review]

**Comment by antonbabenko** [comment]
This PR is included in version 1.6.0 :tada:

**Pull request merged**